Index: gpac/modules/ffmpeg_in/ffmpeg_demux.c
===================================================================
--- gpac.orig/modules/ffmpeg_in/ffmpeg_demux.c
+++ gpac/modules/ffmpeg_in/ffmpeg_demux.c
@@ -531,10 +531,10 @@ static GF_Err FFD_ConnectService(GF_Inpu
 
 	switch (i) {
 	case 0: e = GF_OK; break;
-	case AVERROR_IO: e = GF_URL_ERROR; goto err_exit;
-	case AVERROR_INVALIDDATA: e = GF_NON_COMPLIANT_BITSTREAM; goto err_exit;
-	case AVERROR_NOMEM: e = GF_OUT_OF_MEM; goto err_exit;
-	case AVERROR_NOFMT: e = GF_NOT_SUPPORTED; goto err_exit;
+	case AVERROR(EIO): e = GF_URL_ERROR; goto err_exit;
+	case AVERROR(EINVAL): e = GF_NON_COMPLIANT_BITSTREAM; goto err_exit;
+	case AVERROR(ENOMEM): e = GF_OUT_OF_MEM; goto err_exit;
+	case AVERROR(EILSEQ): e = GF_NOT_SUPPORTED; goto err_exit;
 	default: e = GF_SERVICE_ERROR; goto err_exit;
 	}
 
