--- gpac/applications/dashcast/video_encoder.c~	2018-04-29 18:32:41.000000000 +0200
+++ gpac/applications/dashcast/video_encoder.c	2018-04-29 18:35:58.454087341 +0200
@@ -141,7 +141,7 @@
 	}
 
 	//the global header gives access to the extradata (SPS/PPS)
-	video_output_file->codec_ctx->flags |= CODEC_FLAG_GLOBAL_HEADER;
+	video_output_file->codec_ctx->flags |= AV_CODEC_FLAG_GLOBAL_HEADER;
 
 	video_output_file->vstream_idx = 0;//video_stream->index;
 
