--- gpac-0.7.1/applications/osmo4_wx/Makefile~	2018-09-17 12:18:34.000000000 +0200
+++ gpac-0.7.1/applications/osmo4_wx/Makefile	2018-09-17 12:19:25.217933596 +0200
@@ -46,7 +46,7 @@ SRCS := $(OBJS:.o=.cpp)
 all: $(PROG)
 
 Osmo4$(EXE): $(OBJS)
-	$(CC) -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac $(WX_LFLAGS) $(LINKFLAGS) $(LDFLAGS)
+	$(CC) -o ../../bin/gcc/$@ $(OBJS) -L../../bin/gcc -lgpac $(WX_LFLAGS) `pkg-config --libs gtk+-2.0` -lstdc++ $(LINKFLAGS) $(LDFLAGS)
 
 clean: 
 	rm -f $(OBJS) ../../bin/gcc/$(PROG)
--- gpac-0.7.1/modules/jack/Makefile~	2017-04-26 13:33:17.000000000 +0200
+++ gpac-0.7.1/modules/jack/Makefile	2018-09-17 12:20:29.753236206 +0200
@@ -25,7 +25,7 @@ LIB=gm_jack$(DYN_LIB_SUFFIX)
 all: $(LIB)
 
 $(LIB): $(OBJS)
-	$(CC) $(SHFLAGS) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) $(EXTRALIBS) -L../../bin/gcc -lgpac -L/usr/lib  -ljack
+	$(CC) $(SHFLAGS) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) $(EXTRALIBS) -L../../bin/gcc -lgpac -ljack
 
 clean: 
 	rm -f $(OBJS) ../../bin/gcc/$(LIB)
