--- gpac/modules/amr_float_dec/Makefile.orig	2011-04-21 17:06:51.000000000 +0200
+++ gpac/modules/amr_float_dec/Makefile	2012-06-15 19:19:05.988610730 +0200
@@ -18,19 +18,19 @@
 #common obj
 OBJS=../amr_dec/amr_in.o amr_float_dec.o
 
+AMR_LIBS=
+
 #AMR NB obj
 CFLAGS+=-DGPAC_HAS_AMR_FT
 ifeq ($(CONFIG_AMR_NB_FT), yes)
-OBJS+=./amr_nb_ft/interf_dec.o ./amr_nb_ft/interf_enc.o ./amr_nb_ft/sp_dec.o ./amr_nb_ft/sp_enc.o
+AMR_LIBS+=-lamrnb
 endif
 
 
 #AMR WB obj
 CFLAGS+=-DGPAC_HAS_AMR_FT_WB
 ifeq ($(CONFIG_AMR_WB_FT), yes)
-OBJS+=./amr_wb_ft/dec_acelp.o ./amr_wb_ft/dec_dtx.o ./amr_wb_ft/dec_gain.o ./amr_wb_ft/dec_if.o ./amr_wb_ft/dec_lpc.o ./amr_wb_ft/dec_main.o \
-	./amr_wb_ft/dec_rom.o ./amr_wb_ft/dec_util.o ./amr_wb_ft/enc_acelp.o ./amr_wb_ft/enc_dtx.o ./amr_wb_ft/enc_gain.o ./amr_wb_ft/enc_if.o \
-	./amr_wb_ft/enc_lpc.o ./amr_wb_ft/enc_main.o ./amr_wb_ft/enc_rom.o ./amr_wb_ft/enc_util.o ./amr_wb_ft/if_rom.o 
+AMR_LIBS+=-lamrwb
 endif
 
 
@@ -45,7 +45,7 @@
 
 
 $(LIB): $(OBJS)
-	$(CC) $(SHFLAGS) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) $(EXTRALIBS) -L../../bin/gcc -lgpac
+	$(CC) $(SHFLAGS) $(LDFLAGS) -o ../../bin/gcc/$@ $(OBJS) $(EXTRALIBS) -L../../bin/gcc -lgpac $(AMR_LIBS)
 
 
 %.o: %.c
--- gpac/modules/amr_float_dec/amr_float_dec.c.orig	2012-03-28 14:43:41.000000000 +0200
+++ gpac/modules/amr_float_dec/amr_float_dec.c	2012-06-15 19:20:41.675273381 +0200
@@ -45,12 +45,12 @@
 
 #ifdef GPAC_HAS_AMR_FT
 /*AMR WB*/
-#include "amr_nb_ft/interf_dec.h"
+#include "amrnb/interf_dec.h"
 #endif
 
 #ifdef GPAC_HAS_AMR_FT_WB
 /*AMR WB*/
-#include "amr_wb_ft/dec_if.h"
+#include "amrwb/dec_if.h"
 #endif
 
 #endif
