--- gpac/src/Makefile.orig	2008-12-02 15:01:08.000000000 +0100
+++ gpac/src/Makefile	2011-02-09 16:26:58.340693904 +0100
@@ -174,6 +174,10 @@
 endif
 
 ifeq ($(WANT_PIC),yes)
+LIBGPAC_MCRYPT_PIC=$(LIBGPAC_MCRYPT:.o=.opic)
+LIBGPAC_SCENE_PIC=$(LIBGPAC_SCENE:.o=.opic)
+LIBGPAC_MEDIATOOLS_PIC=$(LIBGPAC_MEDIATOOLS:.o=.opic)
+LIBGPAC_COMPOSITOR_PIC=$(LIBGPAC_COMPOSITOR:.o=.opic)
 OBJSPIC=$(OBJS:.o=.opic)
 else
 OBJSPIC=$(OBJS)
@@ -192,23 +196,19 @@
 lib: $(LIB)
 
 #there's a bunch of warnings in there, get rid of them
-mcrypt: CFLAGS= $(OPTFLAGS) -w -I$(SRC_PATH)/include
-mcrypt: $(LIBGPAC_MCRYPT)
+$(LIBGPAC_MCRYPT) $(LIBGPAC_MCRYPT_PIC): CFLAGS= $(OPTFLAGS) -w -I$(SRC_PATH)/include
 
-scenegraph: CFLAGS += $(SCENEGRAPH_CFLAGS)
-scenegraph: $(LIBGPAC_SCENE)
+$(LIBGPAC_SCENE) $(LIBGPAC_SCENE_PIC): CFLAGS += $(SCENEGRAPH_CFLAGS)
 
-mediatools: CFLAGS += $(MEDIATOOLS_CFLAGS)
-mediatools: $(LIBGPAC_MEDIATOOLS)
+$(LIBGPAC_MEDIATOOLS) $(LIBGPAC_MEDIATOOLS_PIC): CFLAGS += $(MEDIATOOLS_CFLAGS)
 
-compositor: CFLAGS+=$(COMPOSITOR_CFLAGS)
-compositor: $(LIBGPAC_COMPOSITOR)
+$(LIBGPAC_COMPOSITOR) $(LIBGPAC_COMPOSITOR_PIC): CFLAGS+=$(COMPOSITOR_CFLAGS)
 
 %.o: %.c
 	$(CC) $(CFLAGS) -c -o $@ $< 
 
 
-$(LIB): $(LIBGPAC_UTILS) $(LIBGPAC_IETF) $(LIBGPAC_BIFS) $(LIBGPAC_ODF) $(LIBGPAC_LASER) $(LIBGPAC_ISOM) $(LIBGPAC_SCENEMANAGER) $(LIBGPAC_TERMINAL) compositor scenegraph mediatools mcrypt $(OBJSPIC)
+$(LIB): $(OBJS) $(OBJSPIC)
 	ar cr ../bin/gcc/libgpac_static.a $(OBJS)
 	ranlib ../bin/gcc/libgpac_static.a
 	$(CC) $(SHFLAGS) $(LD_SONAME) $(LDFLAGS) -o ../bin/gcc/$@ $(OBJSPIC) $(EXTRALIBS)
