--- gpac-0.7.1/applications/osmozilla/Makefile~	2017-04-26 13:33:17.000000000 +0200
+++ gpac-0.7.1/applications/osmozilla/Makefile	2018-09-17 13:57:38.870996859 +0200
@@ -6,10 +6,6 @@ ifeq ($(CONFIG_WIN32),yes)
 USER_NAME=root
 else
 USER_NAME=$(shell whoami)
-ifeq ($(USER_NAME), root)
-else
-MOZILLA_DIR=local
-endif
 endif
 
 CFLAGS=$(CXXFLAGS) $(XUL_CFLAGS) -I"$(SRC_PATH)/include"
@@ -72,19 +68,8 @@ ifeq ($(CONFIG_WIN32),yes)
 endif
 
 install: 
-ifeq ($(MOZILLA_DIR), local)
-ifeq ($(USER_NAME), root)
-	@echo "*** Root cannot install local mozilla plugins! ***"
-	@echo "*** Exit root mode and reinstall mozilla plugin! ***"
-else
-	$(MAKE) $(LIB)
-	$(INSTALL) -D -m 755 ../../bin/gcc/$(LIB) "$(HOME)/.mozilla/plugins/$(LIB)"
-	$(INSTALL) -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(HOME)/.mozilla/components/nposmozilla.xpt"
-endif
-else
 	$(INSTALL) -D -m 755 ../../bin/gcc/$(LIB) "$(MOZILLA_DIR)/components/$(LIB)"
 	$(INSTALL) -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(MOZILLA_DIR)/components/nposmozilla.xpt"
-endif
 
 uninstall:
 ifeq ($(MOZILLA_DIR), local)
--- gpac-0.7.1/Makefile~	2017-04-26 13:33:17.000000000 +0200
+++ gpac-0.7.1/Makefile	2018-09-17 14:27:00.872110746 +0200
@@ -112,7 +112,7 @@ ifneq ($(CONFIG_WIN32), yes)
 ifneq ($(CONFIG_FFMPEG), no)
 ifneq ($(DISABLE_CORE_TOOLS), yes)
 ifneq ($(DISABLE_AV_PARSERS), yes)
-	if [ -f bin/gcc/DashCast$(EXE_SUFFIX)g ] ; then \
+	if [ -f bin/gcc/DashCast$(EXE_SUFFIX) ] ; then \
 	$(INSTALL) $(INSTFLAGS) -m 755 bin/gcc/DashCast$(EXE_SUFFIX) "$(DESTDIR)$(prefix)/bin" ; \
 	fi
 endif
