--- gpac/applications/osmo4_wx/Makefile.orig	2011-10-21 14:41:56.000000000 +0200
+++ gpac/applications/osmo4_wx/Makefile	2012-06-15 20:20:49.688455299 +0200
@@ -67,7 +67,6 @@
 	      $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/MacOS
 	echo -n 'APPLOsm4' > $(DESTDIR)$(mac_apps)/Osmo4.app/Contents/PkgInfo
 else
-	rm -f wxOsmo4.o
 	mkdir -p $(DESTDIR)$(prefix)/bin
 	install -m 755 $(INSTFLAGS) ../../bin/gcc/Osmo4 "$(DESTDIR)$(prefix)/bin"
 endif
--- gpac/applications/mp4client/Makefile.orig	2011-05-20 19:11:05.000000000 +0200
+++ gpac/applications/mp4client/Makefile	2012-06-15 20:20:27.218456243 +0200
@@ -44,7 +44,7 @@
 clean: 
 	rm -f $(OBJS) ../../bin/gcc/$(PROG)
 
-install: clean
+install:
 	install -m 755 $(INSTFLAGS) ../../bin/gcc/MP4Client "$(DESTDIR)$(prefix)/bin"
 
 uninstall:
