diff -urNp -x '*.orig' gpac-0.8.0.org/applications/Makefile gpac-0.8.0/applications/Makefile
--- gpac-0.8.0.org/applications/Makefile	2019-06-27 12:20:45.000000000 +0200
+++ gpac-0.8.0/applications/Makefile	2021-03-13 09:23:01.300056818 +0100
@@ -33,15 +33,15 @@ V4STUDIODIR=
 INSTDIRS=mp4client
 ifeq ($(CONFIG_XUL),no)
 else
-#INSTDIRS+=osmozilla
-#APPDIRS+=osmozilla
+INSTDIRS+=osmozilla
+APPDIRS+=osmozilla
 endif
 
 #disable due to version incompatibilities
 ifeq ($(USE_WXWIDGETS), yes)
-#APPDIRS+=osmo4_wx
+APPDIRS+=osmo4_wx
 #V4STUDIODIR=V4Studio
-#INSTDIRS+=osmo4_wx
+INSTDIRS+=osmo4_wx
 endif
 
 #MP4BOX_STATIC
