--- gpac/Makefile.orig	2006-03-17 22:16:39.000000000 +0100
+++ gpac/Makefile	2006-03-17 22:17:29.000000000 +0100
@@ -71,8 +71,8 @@
 ifeq ($(CONFIG_WIN32),yes)
 	install -s -m 755 bin/gcc/libgpac.dll $(prefix)/lib
 else
-	install -s -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(prefix)/lib/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
-	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(prefix)/lib/libgpac.$(DYN_LIB_SUFFIX)
+	install -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(libdir)/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
+	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(libdir)/libgpac.$(DYN_LIB_SUFFIX)
 #not sure wether darwin uses ldconfig or not...
 ifeq ($(CONFIG_DARWIN),yes)
 else
--- gpac/Makefile.orig	2006-03-17 22:27:49.000000000 +0100
+++ gpac/Makefile	2006-03-17 22:27:58.000000000 +0100
@@ -64,7 +64,7 @@
 	install -c -s -m 755 bin/gcc/MP4Box "$(prefix)/bin"
 	install -c -s -m 755 bin/gcc/MP42Avi "$(prefix)/bin"
 	$(MAKE) -C applications install
-	install -d "$(moddir)"
+	install -d "$(libdir)/gpac"
 	install -c bin/gcc/*.$(DYN_LIB_SUFFIX) "$(moddir)"
 	rm -f $(moddir)/libgpac.$(DYN_LIB_SUFFIX)
 	rm -f $(moddir)/nposmozilla.$(DYN_LIB_SUFFIX)
--- gpac/applications/osmozilla/Makefile.orig	2006-03-17 22:47:48.000000000 +0100
+++ gpac/applications/osmozilla/Makefile	2006-03-17 22:54:03.000000000 +0100
@@ -5,10 +5,6 @@
 GECKO_PATH=$(SRC_PATH)/extra_lib/include/gecko-sdk
 
 USER_NAME=$(shell whoami)
-ifeq ($(USER_NAME), root)
-else
-MOZILLA_DIR=local
-endif
 
 CFLAGS= $(OPTFLAGS) -Wall -I$(SRC_PATH)/include \
   -I$(GECKO_PATH) \
@@ -71,18 +67,8 @@
 endif
 
 install: $(LIB)
-ifeq ($(MOZILLA_DIR), local)
-ifeq ($(USER_NAME), root)
-	@echo "*** Root cannot install local mozilla plugins! ***"
-	@echo "*** Exit root mode and reinstall mozilla plugin! ***"
-else
-	install -D -m 755 ../../bin/gcc/$(LIB) "$(HOME)/.mozilla/plugins/$(LIB)"
-	install -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(HOME)/.mozilla/components/nposmozilla.xpt"
-endif
-else
-	install -D -m 755 ../../bin/gcc/$(LIB) "$(MOZILLA_DIR)/components/$(LIB)"
-	install -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(MOZILLA_DIR)/components/nposmozilla.xpt"
-endif
+	install -D -m 755 ../../bin/gcc/$(LIB) "$(MOZILLA_DIR)/$(LIB)"
+	install -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(MOZILLA_DIR)/nposmozilla.xpt"
 
 dep: depend
 
--- gpac/Makefile.orig	2006-03-17 22:55:15.000000000 +0100
+++ gpac/Makefile	2006-03-17 22:56:22.000000000 +0100
@@ -65,9 +65,9 @@
 	install -c -s -m 755 bin/gcc/MP42Avi "$(prefix)/bin"
 	$(MAKE) -C applications install
 	install -d "$(libdir)/gpac"
-	install -c bin/gcc/*.$(DYN_LIB_SUFFIX) "$(moddir)"
-	rm -f $(moddir)/libgpac.$(DYN_LIB_SUFFIX)
-	rm -f $(moddir)/nposmozilla.$(DYN_LIB_SUFFIX)
+	install -c bin/gcc/*.$(DYN_LIB_SUFFIX) "$(libdir)/gpac"
+	rm -f $(libdir)/libgpac.$(DYN_LIB_SUFFIX)
+	rm -f $(libdir)/nposmozilla.$(DYN_LIB_SUFFIX)
 ifeq ($(CONFIG_WIN32),yes)
 	install -s -m 755 bin/gcc/libgpac.dll $(prefix)/lib
 else
