--- gpac/applications/osmozilla/nsIOsmozilla.idl~	2006-07-18 18:55:14.000000000 +0200
+++ gpac/applications/osmozilla/nsIOsmozilla.idl	2012-01-22 16:48:30.461393267 +0100
@@ -1,6 +1,6 @@
 #include "nsISupports.idl"
 
-[scriptable, uuid(d2d536a0-b6fc-11d5-9d10-0060b0fbd8bn)]
+[scriptable, uuid(ce32e3ff-36f8-425f-94be-d85b26e634ee)]
 interface nsIOsmozilla : nsISupports {
   void Pause();
   void Play();
