--- gpac/configure.orig	2005-03-31 20:42:08.000000000 +0200
+++ gpac/configure	2005-04-09 04:02:36.000000000 +0200
@@ -588,8 +588,8 @@
 #look for wxWidgets support
 has_wx="no"
 wx_too_old="no"
-wx_cflags=`wx-config --cxxflags`
-wx_lflags=`wx-config --libs`
+wx_cflags=`wx-gtk2-ansi-config --cxxflags`
+wx_lflags=`wx-gtk2-ansi-config --libs`
 
 cat > $TMPCPP << EOF
 #include <wx/wx.h>
@@ -597,7 +597,7 @@
 EOF
 
 if $cc $wx_cflags -o $TMPO $TMPCPP $wx_lflags > /dev/null 2>&1 ; then
-wx_version=`wx-config --version | sed 's/[^0-9]//g'`
+wx_version=`wx-gtk2-ansi-config --version | sed 's/[^0-9]//g'`
 if test "$wx_version" -lt 252 ; then
 wx_too_old="yes"
 else
