--- gpac/Makefile.orig	2008-10-23 17:40:38.000000000 +0200
+++ gpac/Makefile	2011-02-05 07:18:41.996686361 +0100
@@ -59,9 +59,6 @@
 ifeq ($(CONFIG_WIN32),yes)
 	install $(INSTFLAGS) -m 755 bin/gcc/libgpac.dll $(prefix)/$(libdir)
 else
-ifeq ($(DEBUGBUILD),no)
-	$(STRIP) bin/gcc/libgpac.$(DYN_LIB_SUFFIX)
-endif
 ifeq ($(CONFIG_DARWIN),yes)
 	install -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
 	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac.$(DYN_LIB_SUFFIX)
--- gpac/applications/osmozilla/Makefile.orig	2008-11-24 09:54:12.000000000 +0100
+++ gpac/applications/osmozilla/Makefile	2011-02-05 07:20:46.968690551 +0100
@@ -6,10 +6,6 @@
 USER_NAME=root
 else
 USER_NAME=$(shell whoami)
-ifeq ($(USER_NAME), root)
-else
-MOZILLA_DIR=local
-endif
 endif
 
 CFLAGS=$(CPPFLAGS) $(XUL_CFLAGS) -I$(SRC_PATH)/include
@@ -75,19 +71,8 @@
 endif
 
 install: 
-ifeq ($(MOZILLA_DIR), local)
-ifeq ($(USER_NAME), root)
-	@echo "*** Root cannot install local mozilla plugins! ***"
-	@echo "*** Exit root mode and reinstall mozilla plugin! ***"
-else
-	$(MAKE) $(LIB)
-	install -D -m 755 ../../bin/gcc/$(LIB) "$(HOME)/.mozilla/plugins/$(LIB)"
-	install -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(HOME)/.mozilla/components/nposmozilla.xpt"
-endif
-else
-	install -D -m 755 ../../bin/gcc/$(LIB) "$(MOZILLA_DIR)/components/$(LIB)"
-	install -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(MOZILLA_DIR)/components/nposmozilla.xpt"
-endif
+	install -D -m 755 ../../bin/gcc/$(LIB) "$(MOZILLA_DIR)/$(LIB)"
+	install -D -m 755 ../../bin/gcc/nposmozilla.xpt "$(MOZILLA_DIR)/nposmozilla.xpt"
 
 uninstall:
 ifeq ($(MOZILLA_DIR), local)
